src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImage.groovy

```diff
@@ -36,17 +51,42 @@ class DockerSaveImage extends AbstractDockerRemoteApiTask {
 
     DockerSaveImage() {
         useCompression.set(false)
+        onlyIf {
+            images.getOrNull() || image.present
+        }
+    }
+
+    // part of work-around for https://github.com/docker-java/docker-java/issues/1872
+    @CompileDynamic
+    SaveImagesCmd.Exec getExecution() {
+        dockerClient.saveImagesCmd().@execution
     }
 
     @Override
     void runRemoteCommand() {
-        SaveImageCmd saveImageCmd = dockerClient.saveImageCmd(image.get())
-        InputStream image = saveImageCmd.exec()
+        Set<String> images = images.getOrElse([] as Set)
+        if (image.present) {
+            images = images + image.get()
+        }
+        // part of work-around for https://github.com/docker-java/docker-java/issues/1872
+        SaveImagesCmd saveImagesCmd = new SaveImagesCmdImpl(execution) {
+            @Override
+            List<TaggedImage> getImages() {
+                images.collect {
+                    { -> it } as TaggedImage
+                }
+            }
+        }
+        //SaveImagesCmd saveImagesCmd = dockerClient.saveImagesCmd()
```

Comment:
Can you remove the commented out code?

Reply:
Of course, if you wish.
I just thought I leave it in, so once that mentioned bug is resolved, this can just be commented in and the work-around removed.